Microsoft.Toolkit.Services/Services/OneDrive/OneDriveStorageFile.cs

```diff
@@ -40,6 +41,42 @@ public string FileType
             }
         }
 
+        private string _thumbnail;
+
+        /// <summary>
+        /// Gets the smallest available thumbnail for the object
+        /// </summary>
+        public string Thumbnail
+        {
+            get
+            {
+                if (_thumbnail == null)
+                {
+                    GrabThumbnail();
+                }
+                return _thumbnail;
+            }
+        }
+
+        private void GrabThumbnail()
+        {
+            Task.Run(async () =>
+            {
+                try
+                {
+                    var set = await GetThumbnailSetAsync();
+                    if (set != null)
+                    {
+                        _thumbnail = set.Small ?? set.Medium ?? set.Large;
+                    }
+                }
+                catch (Exception)
+                {
+                    _thumbnail = string.Empty;
+                }
+            }).Wait();
```

Comment:
I really don't like doing this, but I can't think of anything that will have the same behavior (lazy loaded async property). @nmetulev ?

Reply:
Good catch. wouldn't this block the UI? Should we just have an asynchronous method to retrieve Thumbnails?

Reply:
I think that is the best approach. But what if the developer wants to bind an image's Source property to it? I hate every since solution that I can think for this.

Reply:
I'd expect the developer to add it to their view model that extends INotifyProperty - seems like the cleanest approach

Reply:
This is lazy load, which is nice, since we don't need the thumbnail for every item.  It will not block the UI while the request is in process, but the get will not return without the result, so you can bind to it without issue.

Reply:
@GraniteStateHacker If you bind directly to it, it will block the UI Thread for as long as that GetThumbnailSetAsync method takes to do the request. Just remove that and add a Task.Delay(10000).
It is lazy loading it, but it is still blocking the UI Thread until it does it (that's what the .Wait() is doing).
This class is a Model, not a ViewModel, so it should not be bind directly to. Properties should return immediately (<50ms). This property should be replaced by a method that returns a Task<string>, this way, the developer can wait for it's completion and create their own VM that handles that.

Reply:
Gotcha, sorry, I was forgetting that Task.Run doesn't keep you on the UI thread, so it doesn't matter that the await is used within it.

Reply:
👍 Once you remove this property @GraniteStateHacker , I think we should be good to go.